lib/smart_proxy_remote_execution_ssh/api.rb

```diff
@@ -1,9 +1,251 @@
+require 'net/ssh'
+require 'forwardable'
+
+# When hijacking the socket of a TLS connection, we get a
+# OpenSSL::SSL::SSLSocket or Puma::MiniSSL::Socket, which don't behave
+# the same as real IO::Sockets.  We need to add recv and send for the
+# benefit of the Net::SSH::BufferedIo mixin, and closed? for our own
+# convenience.
+
+
 module Proxy::RemoteExecution
   module Ssh
+    class BufferedSocket
```

Comment:
@mvollmer I've pushed the commit that replaces the monkey-patching with wrapper classes. This gives me more confidence about this code not affecting anything else

Reply:
I have tested it in my three-machine setup, and didn't find any issues.  The case where we need to wait in order to avoid returning "" to Net::SSH still happens and is still handled correctly.